flexmeasures/data/schemas/generic_assets.py

```diff
@@ -151,16 +151,24 @@ class Meta:
 class GenericAssetIdField(MarshmallowClickMixin, fields.Int):
     """Field that deserializes to a GenericAsset and serializes back to an integer."""
 
-    @with_appcontext_if_needed()
-    def _deserialize(self, value, attr, obj, **kwargs) -> GenericAsset:
+    def __init__(self, status_if_not_found: int | None = None, *args, **kwargs):
```

Comment:
For the argument `status_if_not_found`, should we use an Enum or an object like [HTTPStatus](https://docs.python.org/3/library/http.html#http.HTTPStatus)? It's a bit more verbose but I think it would look nicer to see something like `HTTPStatus.NOT_FOUND`.